examples/list_box_model/list_box_row/imp.rs

```diff
@@ -0,0 +1,154 @@
+                    glib::ParamFlags::READWRITE | glib::ParamFlags::CONSTRUCT,
+                ),
+                ParamSpecObject::new(
+                    "parent-window",
+                    "Parent Window",
+                    "Parent Window",
+                    gtk::Window::static_type(),
+                    glib::ParamFlags::READWRITE | glib::ParamFlags::CONSTRUCT,
+                ),
+            ]
+        });
+        PROPERTIES.as_ref()
+    }
+
+    fn set_property(&self, _obj: &Self::Type, _id: usize, value: &Value, pspec: &ParamSpec) {
+        match pspec.name() {
+            "rowdata" => {
+                let row_data = value.get().unwrap();
+                self.row_data.replace(row_data);
+            }
+            "parent-window" => {
+                let parent_window = value.get().unwrap();
+                self.parent_window.replace(parent_window);
+            }
+            _ => unimplemented!(),
+        }
+    }
+
+    fn property(&self, _obj: &Self::Type, _id: usize, pspec: &ParamSpec) -> Value {
+        match pspec.name() {
+            "rowdata" => self.row_data.borrow().to_value(),
+            "parent-window" => self.parent_window.borrow().to_value(),
+            _ => unimplemented!(),
+        }
+    }
+
+    fn constructed(&self, obj: &Self::Type) {
+        let item = self.row_data.borrow();
+        let item = item.as_ref().cloned().unwrap();
+
+        let hbox = gtk::Box::new(gtk::Orientation::Horizontal, 5);
+
+        // Create the label and spin button that shows the two values
+        // of the item. We bind the properties for the two values to the
+        // corresponding properties of the widgets so that they are automatically
+        // updated whenever the item is changing. By specifying SYNC_CREATE the
+        // widget will automatically get the initial value of the item set.
+        //
+        // In case of the spin button the binding is bidirectional, that is any
+        // change of value in the spin button will be automatically reflected in
+        // the item.
+        let label = gtk::Label::new(None);
+        item.bind_property("name", &label, "label")
+            .flags(glib::BindingFlags::DEFAULT | glib::BindingFlags::SYNC_CREATE)
+            .build();
+        hbox.append(&label);
+
+        let spin_button = gtk::SpinButton::with_range(0.0, 100.0, 1.0);
+        item.bind_property("count", &spin_button, "value")
+            .flags(
+                glib::BindingFlags::DEFAULT
+                    | glib::BindingFlags::SYNC_CREATE
+                    | glib::BindingFlags::BIDIRECTIONAL,
+            )
+            .build();
+        hbox.append(&spin_button);
+
+        // When the edit button is clicked, a new modal dialog is created for editing
+        // the corresponding row
+        let edit_button = gtk::Button::with_label("Edit");
+        edit_button.connect_clicked(clone!(@strong item, @strong self.parent_window as window => move |_| {
```

Comment:
```suggestion
        edit_button.connect_clicked(clone!(@weak item, @weak self.parent_window as window => move |_| {
```

Why not? I think you're otherwise creating a reference cycle here.

Reply:
Honestly this specific part of clone! is a bit magicky to me, but switching to `@weak` still works so I think that's fine